pennylane/capture/capture_qnode.py

```diff
@@ -0,0 +1,162 @@
+    qnode_prim.multiple_results = True
+
+    @qnode_prim.def_impl
+    def _(*args, shots, device, qnode_kwargs, qfunc_jaxpr):
+        def qfunc(*inner_args):
+            return jax.core.eval_jaxpr(qfunc_jaxpr.jaxpr, qfunc_jaxpr.consts, *inner_args)
+
+        qnode = qml.QNode(qfunc, device, **qnode_kwargs)
+        return qnode._impl_call(*args, shots=shots)  # pylint: disable=protected-access
+
+    # pylint: disable=unused-argument
+    @qnode_prim.def_abstract_eval
+    def _(*args, shots, device, qnode_kwargs, qfunc_jaxpr):
+        mps = qfunc_jaxpr.out_avals
+        return _get_shapes_for(*mps, shots=shots, num_device_wires=len(device.wires))
+
+    return qnode_prim
+
+
+# pylint: disable=protected-access
+def _get_device_shots(device) -> "qml.measurements.Shots":
+    if isinstance(device, qml.devices.LegacyDevice):
+        if device._shot_vector:
+            return qml.measurements.Shots(device._raw_shot_sequence)
+        return qml.measurements.Shots(device.shots)
+    return device.shots
+
+
+def qnode_call(qnode: "qml.QNode", *args, **kwargs) -> "qml.typing.Result":
+    """A capture compatible call to a qnode.
+
+    Args:
+        qnode (QNode): a qnode
+        *args: the arguments the qnode is called with
+
+    Keyword Args:
+        Any keyword arguments accepted by the quantum function
+
+    Returns:
+        qml.typing.Result: the result of a qnode execution
+
+    **Example:**
+
+    .. code-block:: python
+
+        @qml.qnode(qml.device('lightning.qubit', wires=1))
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.Z(0)), qml.probs()
+
+        def f(x):
+            expval_z, probs = circuit(np.pi * x, shots=50)
+            return 2*expval_z + probs
+
+        jaxpr = jax.make_jaxpr(f)(0.1)
+        print("jaxpr: \n", jaxpr)
+
+        res = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 0.7)
+        print("\nresult: \n", res)
+
+
+    .. code-block:: none
+
+        jaxpr:
+        { lambda ; a:f32[]. let
+            b:f32[] = mul 3.141592653589793 a
+            c:f32[] d:f32[2] = qnode[
+            device=<lightning.qubit device (wires=1) at 0x30755b3d0>
+            qfunc_jaxpr={ lambda ; e:f32[]. let
+                _:AbstractOperator() = RX[n_wires=1] e 0
+                f:AbstractOperator() = PauliZ[n_wires=1] 0
+                g:AbstractMeasurement(n_wires=None) = expval_obs f
+                h:AbstractMeasurement(n_wires=0) = probs_wires
+                in (g, h) }
+            qnode_kwargs={'diff_method': 'best', 'grad_on_execution': 'best', 'cache': False, 'cachesize': 10000, 'max_diff': 1, 'max_expansion': 10, 'device_vjp': False}
+            shots=Shots(total=50)
+            ] b
+            i:f32[] = mul 2.0 c
+            j:f32[2] = add i d
+        in (j,) }
+
+        result:
+        [Array([-1.08, -0.64], dtype=float32)]
+
+
+    """
+    shots = kwargs.pop("shots", _get_device_shots(qnode.device))
+    shots = qml.measurements.Shots(shots)
```

Comment:
This always executes `_get_device_shots`, right? Maybe we want to go the manual route here:

```suggestion
    if "shots" in kwargs:
        shots = qml.measurements.Shots(kwargs.pop("shots"))
    else:
        shots = _get_device_shots(qnode.device)
```

Reply:
What about this? Would you like to stick to calling `_get_device_shots` in any case?